### DISCUSSION TITLE
Issue with Nexmon Kernel Patch Compilation - No rule to make target '/firmwares/bcm43455c0/7_45_154/definitions.mk'

### DISCUSSION THREAD

**Discussion by Hysome** [seemoo-lab/nexmon#570] [Q&A]
I am currently working on compiling the Nexmon kernel patch for my Raspberry Pi 4B with the bcm43455c0 chipset. However, I have encountered an error during the compilation process, and I am seeking some assistance to resolve it.

The specific error I encountered is as follows:

**_Makefile:4: /firmwares/bcm43455c0/7_45_154/definitions.mk: No such file or directory
make: *** No rule to make target '/firmwares/bcm43455c0/7_45_154/definitions.mk'. Stop._**

I have ensured that the definitions.mk file and other firmware files are present in the firmwares/bcm43455c0/7_45_154/ directory. I also tried setting the FIRMWARE_DIR variable manually to point to the correct location, but the issue persists.

Here are the steps I followed:

1. Cloned the Nexmon repository and checked out the appropriate branch.
2. Downloaded and extracted the firmware files.
3. Navigated to the nexmon/patches/bcm43455c0/7_45_154/nexmon directory.
Attempted to run make, but encountered the error mentioned above.
I have also tried cleaning the build and updating the firmware submodule, but it did not resolve the issue.

Could you please provide some guidance on how to troubleshoot this error and successfully compile the kernel patch for my Raspberry Pi 4B with bcm43455c0 chipset?

**Comment by matthiasseemoo** [comment]
read the readme again and follow every step including the one calling setup_env.sh

**Reply by Hysome** [reply]
Thank you @matthiasseemoo, I tried that still doesn't fix
